dev/build_prevalidation.py

```diff
@@ -57,18 +57,24 @@ def __record_halyard_nightly_version(version_bump, options):
   check_run_quick('gsutil rsync -r -d {remote_uri} {local_bucket}'
                   .format(remote_uri=bucket_uri, local_bucket=local_bucket_name))
   hal_version = version_bump.version_str.replace('version-', '')
-  new_hal_nightly_entry = ('{version}-{build}: {commit}'
-                           .format(version=hal_version, build=build_number, commit=version_bump.commit_hash))
+  full_hal_version = '{version}-{build}'.format(version=hal_version, build=build_number)
+  new_hal_nightly_entry = ('{full_hal_version}: {commit}'
+                           .format(full_hal_version=full_hal_version, commit=version_bump.commit_hash))
   nightly_entry_file = '{0}/nightly-version-commits.yml'.format(local_bucket_name)
   with open(nightly_entry_file, 'a') as nef:
     nef.write('{0}\n'.format(new_hal_nightly_entry))
   # Now sync the local dir with the bucket again after the update.
   check_run_quick('gsutil rsync -r -d {local_bucket} {remote_uri}'
                   .format(remote_uri=bucket_uri, local_bucket=local_bucket_name))
 
+  with open(options.hal_version_under_test_file_path, 'w') as hal_version_file:
```

Comment:
'w' should create a new [file each time](https://docs.python.org/2/tutorial/inputoutput.html#reading-and-writing-files).